include/xdpw.h

```diff
@@ -7,23 +7,29 @@
 #elif HAVE_ELOGIND
 #include <elogind/sd-bus.h>
 #endif
-#include "logger.h"
-#include "screencast.h"
+
+#include "screencast_common.h"
 
 struct xdpw_state {
+	struct wl_list xdpw_sessions;
 	sd_bus *bus;
 	struct wl_display *wl_display;
 	struct pw_loop *pw_loop;
-
-	struct screencast_context screencast;
+	struct xdpw_screencast_context screencast;
+	enum source_types screencast_source_types;
```

Comment:
enums should only take values enumerated in the definition. For bitfields, this isn't the case. Prefer something like:

```c
uint32_t screencast_source_types; // bitfield of enum source_types
```